### PR TITLE
ref(dashboard-layout): Merge dndkit and grid dashboard files

### PR DESCRIPTION
**Pull request by narsaynorath**
Most of this code is bringing stuff over from the grid layout dashboard file and merging it into the existing one.

The logic splits in the necessary handler functions by checking for the `dashboard-grid-layout` flag. The exceptions were the handlers for adding, deleting, and updating methods because I believe the grid layout versions will work with the existing ones and they don't rely on the index.

❓ What should I do with the tests? Should I merge the files as well? It's going to become a really long test file.

### CODE REVIEW

File: static/app/views/dashboardsV2/dashboard.tsx [hunk 1]
```diff
@@ -1,10 +1,17 @@
+import 'react-grid-layout/css/styles.css';
```
Comment (narsaynorath) [inline static/app/views/dashboardsV2/dashboard.tsx:1]: Wondering if there's a pattern we can use to only import these if the flag is set 🤔

**Review by shruthilayaj — APPROVED** [review]
Changes look good to me, testing with the flag on and off also look good! The outstanding thing is the tests, we discussed that you'd try to remove the test cases under gridLayout that weren't really testing gridLayout stuff and just there to prevent regressions.

One bug I noticed (that's currently also on master so not to do with this PR) is when I "Create Dashboard" with grid layout enabled, add a widget, edit the title and click Save and Finish it shows the "You have unsaved changes, are you sure you want to leave?" pop up. This can be fixed in a follow up! There's another bug with the "Create Dashboard" flow and Widget Library which I've put up a fix for [here](https://github.com/getsentry/sentry/pull/30816)

File: static/app/views/dashboardsV2/dashboard.tsx [hunk 2]
```diff
@@ -133,26 +197,41 @@ class Dashboard extends Component<Props> {
...
+    }
+    this.props.onUpdate([...this.props.dashboard.widgets, newWidget]);
   };
 
-  handleUpdateComplete = (index: number) => (nextWidget: Widget) => {
+  handleUpdateComplete = (prevWidget: Widget) => (nextWidget: Widget) => {
     const nextList = [...this.props.dashboard.widgets];
-    nextList[index] = nextWidget;
+    const updateIndex = nextList.indexOf(prevWidget);
+    nextList[updateIndex] = {...nextWidget, tempId: prevWidget.tempId};
```
[the remark this answers is not in the document]
Comment (narsaynorath) [reply static/app/views/dashboardsV2/dashboard.tsx:210]: Yeah, if it's undefined it'll just transfer over as undefined

File: static/app/views/dashboardsV2/dashboard.tsx (hunk 1 above)
Comment (narsaynorath) [reply static/app/views/dashboardsV2/dashboard.tsx:1]: Going to punt on this. I tried adding the import step to `componentWillMount` where I can check the flag, but I was getting TS errors saying the module, or its types, couldn't be found. At this point I figured trying to lazy load will overcomplicate things

**Pull request merged**